crates/secrets-nats-kv/src/types.rs

```diff
@@ -2,6 +2,14 @@ use bytes::Bytes;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
+#[derive(Serialize, Deserialize, Default)]
+pub struct PutSecretRequest {
+    pub key: String,
+    pub version: String,
+    pub string_secret: Option<String>,
+    pub binary_secret: Option<Vec<u8>>,
+}
```

Comment:
This seems familiar, we can't re-use it from somewhere? 

ahh it's more like a serialization of the `SecretValue` + `key` + `version`, which is why it seems familiar. If this is something we'd use from more than one secrets provider it might be nice to move it up to a shared package

Reply:
For bit of context, all of the other secrets providers that we've implemented or considered implementing so far have their own set of management APIs and tooling, NATS KV just happens to be the one we built on top of NATS, so these are required for NATS, but not for any of the other providers.

Reply:
Ah thanks -- I guess using `SercretValue` as part of this struct isn't a huge requirement either, :rocket: